daemon/container_operations.go

```diff
@@ -980,6 +980,9 @@ func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName
 	if endpointConfig == nil {
 		endpointConfig = &networktypes.EndpointSettings{}
 	}
+	container.Lock()
+	defer container.Unlock()
```

Comment:
The only long operation here that does not warrant a lock `FindNetwork`. 
So I could unlock during `FindNetwork` but will have to recheck the state after re-locking.

Everything else we need to prevent any state changes to the container while the operations are performed.